includes/wizards/class-onboarding-wizard.php

```diff
@@ -0,0 +1,339 @@
+	}
+
+	/**
+	 * Save location info.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Boolean success.
+	 */
+	public function api_save_location( $request ) {
+		$params   = $request->get_params();
+		$defaults = [
+			'countrystate' => '',
+			'address1'     => '',
+			'address2'     => '',
+			'city'         => '',
+			'postcode'     => '',
+			'currency'     => '',
+		];
+		$args     = wp_parse_args( $params, $defaults );
+
+		update_option( 'woocommerce_store_address', $args['address1'] );
+		update_option( 'woocommerce_store_address_2', $args['address2'] );
+		update_option( 'woocommerce_store_city', $args['city'] );
+		update_option( 'woocommerce_default_country', $args['countrystate'] );
+		update_option( 'woocommerce_store_postcode', $args['postcode'] );
+		update_option( 'woocommerce_currency', $args['currency'] );
+
+		// @todo when is the best time to do this?
+		$this->set_smart_defaults();
+
+		return rest_ensure_response( true );
+	}
+
+	/**
+	 * Get Stripe settings.
+	 *
+	 * @return WP_REST_Response containing info.
+	 */
+	public function api_get_stripe_settings() {
+		$defaults = [
+			'enabled'            => false,
+			'testMode'           => false,
+			'publishableKey'     => '',
+			'secretKey'          => '',
+			'testPublishableKey' => '',
+			'testSecretKey'      => '',
+		];
+		$gateways = WC_Payment_Gateways::instance()->payment_gateways();
+
+		if ( ! isset( $gateways['stripe'] ) ) {
+			return rest_ensure_response( $defaults );
+		}
+
+		$stripe   = $gateways['stripe'];
+		$settings = [
+			'enabled'            => 'yes' === $stripe->get_option( 'enabled', false ) ? true : false,
+			'testMode'           => 'yes' === $stripe->get_option( 'testmode', false ) ? true : false,
+			'publishableKey'     => $stripe->get_option( 'publishable_key', '' ),
+			'secretKey'          => $stripe->get_option( 'secret_key', '' ),
+			'testPublishableKey' => $stripe->get_option( 'test_publishable_key', '' ),
+			'testSecretKey'      => $stripe->get_option( 'test_secret_key', '' ),
+		];
+
+		return rest_ensure_response( $settings );
+	}
+
+	/**
+	 * Save Stripe settings.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Boolean success.
+	 */
+	public function api_save_stripe_settings( $request ) {
+		$params   = $request->get_params();
+		$defaults = [
+			'enabled'            => false,
+			'testMode'           => false,
+			'publishableKey'     => '',
+			'secretKey'          => '',
+			'testPublishableKey' => '',
+			'testSecretKey'      => '',
+		];
+		$args     = wp_parse_args( $params, $defaults );
+
+		$gateways = WC_Payment_Gateways::instance()->payment_gateways();
+		if ( ! isset( $gateways['stripe'] ) ) {
+			if ( $args['enabled'] ) {
+				// Stripe is not installed and we want to use it. Install/Activate/Initialize it.
+				Plugin_Manager::activate( 'woocommerce-gateway-stripe' );
+				do_action( 'plugins_loaded' );
```

Comment:
I don't believe it always need to be called. It just needed to be called this time because the Stripe gateway hooks into that action to register the gateway with WooCommerce, and it needed to be registered with WC before we could use it. It's easy enough to refactor in the future if we find ourselves duplicating it a lot.